### ISSUE TITLE
Allow more than Clicks to trigger actions

### ISSUE THREAD

**Issue by bontaq** [purview-framework/purview#6]
Right now there's just
```
data Attributes where
  OnClick :: ToJSON a => a -> Attributes
```
and in `src/Wrapper.hs` only clicks are dealt with.

There are a lot of different kinds of events (https://www.w3schools.com/jsref/dom_obj_event.asp), even just focusing on `MouseEvent`s.  There should be a general way for users to listen for any of these events on a component and have it send an action.  

Making them all built-in would get messy and hinder creativity (if someone wanted to use an event that hasn't been added yet), though the most common ones could be included by default (onClick, onBlur, a few more).

**Comment by bontaq** [comment]
I'm actually gonna leave this for the release after the initial one, I think it's cool and much needed but gotta trim stuff somewhere!

**Comment by bontaq** [comment]
For the initial release, don't worry about passing up information from the event.  That can come later.

**Comment by bontaq** [comment]
Fixed for now by #62
